### PR TITLE
Add oauth2 PKCE test case

### PR DESCRIPTION
**Pull request by Thumimku**
Public Issue : https://github.com/wso2/product-is/issues/21313

Currently we don't have a test suite to test PKCE scenarios in integration test nodule, but we have one in scenario test which cover legit customer scenario which is not sufficient for testing edge cases and frequent changes in oauth repo. Hence this PR will address above isue.

Verified scenarios are,

- Default App
  - PKCE "S256" with valid verifier
  - PKCE "S256" with invalid verifier
  - PKCE "S256" without verifier
  - PKCE  with invalid challenge method
  - PKCE  with plain challenge method

- Mandate PKCE for the App withot allowing Plain challenge method
  - PKCE "S256" with valid verifier
  - PKCE "S256" with invalid verifier
  - PKCE "S256" without verifier
  - PKCE  with invalid challenge method
  - PKCE  with plain challenge method

- Mandate PKCE for the App with allowing Plain challenge method
  - PKCE "S256" with valid verifier
  - PKCE "S256" with invalid verifier
  - PKCE "S256" without verifier
  - PKCE  with invalid challenge method
  - PKCE  with plain challenge method

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2PKCETestCase.java [hunk 1]
```diff
@@ -0,0 +1,588 @@
...
+    private Lookup<CookieSpecProvider> cookieSpecRegistry;
+    private RequestConfig requestConfig;
+    private CloseableHttpClient client;
+    private String sessionDataKey;
+    private String sessionDataKeyConsent;
+    private AuthorizationCode authorizationCode;
+    private String applicationId;
+    private String pkceVerifier;
+
+
```
Comment (asha15) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2PKCETestCase.java:93]: Remove the additional line.

Comment (Thumimku) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2PKCETestCase.java]: fixed with https://github.com/wso2/product-is/pull/21389/commits/c19fd77e6823d7dcc4c235157aa968cb2a88a7a8

**Review by asha15 — APPROVED** [review]
